test/jdk/java/awt/im/bug4490692.java

```diff
@@ -0,0 +1,112 @@
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+import java.awt.BorderLayout;
+import java.awt.Container;
+import java.awt.event.KeyEvent;
+import java.awt.event.KeyListener;
+import javax.swing.JFrame;
+import javax.swing.JTextField;
+
+/*
+ * @test
+ * @bug 4490692
+ * @summary [Linux] Test for KEY_PRESS event for accented characters.
+ * @requires (os.name == "linux")
+ * @library /java/awt/regtesthelpers
+ * @build PassFailJFrame
+ * @run main/manual bug4490692
+ */
+
+public class bug4490692 {
+    private static final String INSTRUCTIONS = """
+            This test is for unix platforms only.
+            Before the test, you need to modify the keyboard mapping for
+            Tab by issuing the following command:
+
+            xmodmap -e 'keycode 23 = aacute'  (this is for Linux)
+            xmodmap -e 'keycode 60 = aacute'  (this is for Solaris Sparc)
+
+            This command lets you type 'a with acute (à)' character when
+            you press 'Tab' key. After the test, please DO NOT fail to
+            restore the original key mapping by doing the following.
+
+            xmodmap -e 'keycode 23 = Tab'  (this is for Linux)
+            xmodmap -e 'keycode 60 = Tab'  (this is for Solaris Sparc)
+
+            CASE 1: This is a manual check and for SOLARIS SPARC keyboard
+            only. Check whether the key sequence ("Compose", "a", " ' ")
+            generates a-acute character in en_US locale.
+
+            CASE 2: This step is automated and applicable for both
+            keyboards - LINUX & SOLARIS SPARC.
+            When Tab key is pressed it should generate a-acute (à)
+            character, this test automatically passes if the correct character
+            is generated on keypress else fails.
+            """;
+
+    public static void main(String[] args) throws Exception {
+        PassFailJFrame.builder()
+                .title("Test Instructions")
+                .instructions(INSTRUCTIONS)
+                .rows((int) INSTRUCTIONS.lines().count() + 2)
+                .columns(40)
+                .testUI(() -> new TestFrame("Test Accented Chars"))
+                .build()
+                .awaitAndCheck();
+    }
+}
+
+class TestFrame extends JFrame implements KeyListener {
+    JTextField text;
+
+    TestFrame(String title) {
+        super(title);
+        text = new JTextField();
+        text.addKeyListener(this);
+        Container c = getContentPane();
+        c.setLayout(new BorderLayout());
+        c.add(text, BorderLayout.CENTER);
+        setSize(300, 200);
+    }
+
+    @Override
+    public void keyPressed(KeyEvent e) {
+        if (e.getKeyCode() == 23 || e.getKeyCode() == 60) {
```

Comment:
>  CASE 2: This step is automated and applicable for both
keyboards - LINUX & SOLARIS SPARC.
When Tab key is pressed it should generate a-acute (à)
character, this test automatically passes if the correct character
is generated on keypress else fails.

Unfortunately this check doesn't work for me:
`java.awt.event.KeyEvent[KEY_PRESSED,keyCode=0,keyText=Unknown keyCode: 0x0,keyChar='á',keyLocation=KEY_LOCATION_UNKNOWN,rawCode=23,primaryLevelUnicode=225,scancode=0,extendedKeyCode=0x10000e1] on javax.swing.JTextField...`

The `keycode` is 0, `rawCode` is desired `23`, but we can't get it it unless parsing the output of `e.toString()`

The `e.getKeyChar() == 0x00e1` check is still valid thought.

Reply:
@azvegint Thanks for checking it. Update the code & added the rawCode check now. Can you please re-review it.

Reply:
Now it works for the tab case, thank you.